app/Resources/all_languages.json

```diff
@@ -87,7 +87,8 @@
     "date_format_full": "Y-m-d H:i:s",
     "is_rtl": "0",
     "language_code": "ca-es",
-    "locale": "ca-ES"
+    "locale": "ca-ES",
+    "flag": "catalonia"
```

Comment:
Any reason to have `"flag"` only for `ca-es` and `en-ud`?

Reply:
Yes. By default, PrestaShop maps the country part of the locale to the country flag, which is in itself an approximation, but works fairly well for regional variations when they belong to a country, like Canadian French or Argentinian Spanish. However, in the case of ca-ES the country code is ES which would be mapped to the flag of Spain, but we'd want instead to use the Catalonian flag. Same goes for UD which is not a country in this case, but a variation.